packages/dataviews/src/validation.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Internal dependencies
+ */
+import { normalizeFields } from './normalize-fields';
+import type { Field, Form } from './types';
+
+export function isItemValid< Item >(
```

Comment:
This method just ports the existing logic into the new place, there's no behavioral changes.

In follow-up PRs we should discuss all the validation details iteratively: how a form declares the required fields (or is it the field itself?), etc. Those conversations shouldn't block this PR from landing.